install-deps.sh

```diff
@@ -21,18 +21,22 @@ export LC_ALL=C # the following is vulnerable to i18n
 
 function ensure_decent_gcc_on_deb {
     # point gcc to the one offered by distro if the used one is different
-    local old=$(readlink -e /usr/bin/x86_64-linux-gnu-gcc | cut -d'-' -f2)
+    local old=$(readlink -e /usr/bin/gcc | cut -d'-' -f2)
     local new=$1
-    if dpkg --compare-versions $old eq $new; then
+    if [ ! -f /usr/bin/gcc-${old} ]; then
+        $SUDO rm -f /usr/bin/gcc-${old}
+        $SUDO ln -sf /usr/bin/gcc-${new} /usr/bin/gcc
```

Comment:
@tchaikov If `/usr/bin/gcc-${new}` doesn't exist (for whatever reason), this will just create a broken symlink?

If gcc 7 is installed on the Trusty/Xenial system, there are two possibilities - either it is coexisting with the distro version, or it was installed as an upgrade. In the latter case the distro version would be gone and `/usr/bin/gcc-${new}` would not exist.

Reply:
>  If `/usr/bin/gcc-${new}` doesn't exist (for whatever reason), this will just create a broken symlink?

@smithfarm yes, but please see https://github.com/ceph/ceph/pull/19521/files#diff-47a21b3706c13e08943e223c12323aa1R63 . the only reason is that `apt-get install -y dpkg-dev gcc` fails, but see https://github.com/ceph/ceph/pull/19521/files#diff-47a21b3706c13e08943e223c12323aa1R1 , and per `help set`

>       -e  Exit immediately if a command exits with a non-zero status.

this should not happen.

> or it was installed as an upgrade

gcc-7 does not conflict or break gcc-5 or gcc-4.8. so it's not an upgrade.

> In the latter case the distro version would be gone and `/usr/bin/gcc-${new}` would not exist.

this is not a valid reasoning, because its hypothesis is false.